lib/kvdb/kvdb_rparams.c

```diff
@@ -1260,10 +1577,49 @@ kvdb_rparams_defaults()
 }
 
 merr_t
-kvdb_rparams_resolve(struct kvdb_rparams *params, const char *home)
+kvdb_rparams_get(
+    const struct kvdb_rparams *const params,
+    const char *const                param,
+    char *const                      buf,
+    const size_t                     buf_sz,
+    size_t *const                    needed_sz)
 {
-    assert(params);
-    assert(home);
+    const struct params p = { .p_params = { .as_kvdb_rp = params }, .p_type = PARAMS_KVDB_RP };
 
-    return 0;
+    return param_get(&p, pspecs, NELEM(pspecs), param, buf, buf_sz, needed_sz);
+}
+
+merr_t
+kvdb_rparams_set(
+    const struct kvdb_rparams *const params,
+    const char *const                param,
+    const char *const                value)
+{
+    merr_t                 err;
+    int                    rc;
+    static pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
+
+    if (!params || !param || !value)
+        return merr(EINVAL);
+
+    const struct params p = { .p_params = { .as_kvdb_rp = params }, .p_type = PARAMS_KVDB_RP };
+
+    rc = pthread_mutex_lock(&lock);
```

Comment:
Use mutex_lock/mutex_unlock?

Reply:
I didn't use mutex because I don't really find it useful. I don't see any gain over pthread_mutex_t. Ideally I would like to get rid of struct mutex.

Reply:
I'm fine with it if that's the plan.. Just pointed out the mutex_lock/unlock wrapper as it takes care of asserting on the return value.

Reply:
The bigger problem is that, whether we like it or not, we do have a mutex abstraction.  We should either agree to dismantle it and start writing code like above, or stick with it.  The upside to the abstraction is that we can change it if we want and all uses cases benefit (e.g. say we want to write our own futex-based mutex w/o all the pthread overhead).  Case in point, I can write a version of our spinlock_t leveraging C11 atomics that has half the latency of pthread_spinlock_t lock/unlock.  And I actually plan to one of these days.  But I wouldn't even consider it if I had to address a hundred call sites, each with their own idiosyncratic approach (although maybe a bad example since there aren't very many call sites left.  But there are *many* mutex call sites...)